app/main.py

```diff
@@ -1,2 +1,68 @@
 class Dictionary:
-    pass
+
+    def __init__(self):
+        self.key = None
+        self.value = None
+        self.capacity = 8
+        self.threshold = 5
+        self.non_empty = 0
+        self.dict = [None for _ in range(self.capacity)]
+
+    def __setitem__(self, key, value):
+        self.key = key
+        self.value = value
+        self.hash = self.__hash__()
+        self.add_to_dict()
+
+    def __getitem__(self, key):
+        for item in self.dict:
+            if type(item) is list and item[0] == key:
+                return item[1]
+        raise KeyError
+
+    def __repr__(self):
+        return f"{{{self.key}: {self.value}}}"
+
+    def choose_index(self):
+        index = self.hash % self.capacity
+        if not any(self.dict):
+            self.dict[index] = [self.key, self.value, self.hash]
+            self.non_empty += 1
+            return
+        for index in range(len(self.dict)):
+            if type(self.dict[index]) is list:
+                if self.dict[index][0] == self.key:
+                    self.dict[index][1] = self.value
+                    return
+        if self.dict[index]:
+            for index_after in range(index + 1, self.capacity):
```

Comment:
This loop takes O(n) time for every search of the missing key. You should optimize it :(